eth/stagedsync/stage_headers.go

```diff
@@ -566,13 +566,24 @@ func verifyAndSaveNewPoSHeader(
 		// Side chain or something weird
 		// TODO(yperbasis): considered non-canonical because some missing headers were downloaded but not canonized
 		// Or it's not a problem because forkChoice is updated frequently?
-
+		if cfg.memoryOverlay {
+			var checked bool
+			if checked, err = cfg.hd.ValidatePayload(tx, header, body, false, cfg.execPayload); err != nil {
```

Comment:
I do not think that it is as trivial as that, we run a stageloop after all and we would have to check one by one for each error one is from the database and which one is not. i propose a KISS(keep it simple stupid) approach for now because that sounds like a giant headache... regarding returning checked/unchecked/invalid, that is possible

Reply:
actually that is doeable to some extent, just kidding

Reply:
I made it more elegant, could you review again?